src/com/xilinx/rapidwright/edif/EDIFNet.java

```diff
@@ -131,12 +128,35 @@ public EDIFPortInst createPortInst(EDIFPort port, int index, EDIFCellInst cellIn
 		return new EDIFPortInst(port, this, index, cellInst);
 	}
 	
-	public Map<String,EDIFPortInst> getPortInstMap(){
-		return portInsts == null ? Collections.emptyMap() : portInsts;
+	/**
+	 * Creates a new map of all the EDIFPortInst objects stored on this net.  The new map
+	 * contains a copy of EDIFPortInsts available at the time of invocation as returned from 
+	 * {@link #getPortInstList()}.      
+	 * @return A map of EDIFPortInst names ({@link EDIFPortInst#getName()} to the corresponding objects.
+	 */
+	public Map<String, EDIFPortInst> getPortInstMap(){
+	    if(portInsts == null) return Collections.emptyMap();
+	    HashMap<String, EDIFPortInst> map = new HashMap<>();
+	    for(EDIFPortInst e : getPortInstList()) {
+	        map.put(e.getFullName(), e);
+	    }
+	    return map;
```

Comment:
Is this worth keeping around? GH is telling me `EDIFNet` is the only user, and all it seems to do is build this map, do a single lookup, then throw the map away.

Reply:
I think I meant to tag it as deprecated